src/main/java/net/pms/configuration/sharedcontent/SharedContentConfiguration.java

```diff
@@ -147,6 +150,13 @@ public static void updateSharedContent(SharedContentArray values, boolean save)
 					PMS.get().resetRenderersRoot();
 				}
 				updated = true;
+
+				if (wasFolderUpdate) {
+					// Rescan to add/remove Media Library content
+					if (CONFIGURATION.getUseCache() && !LibraryScanner.isScanLibraryRunning()) {
+						LibraryScanner.scanLibrary();
```

Comment:
Shouldn't the current running thread be stopped, and a new library scan started ?

Reply:
Good point, that's changed now